challenges/ui/cakes_shop/luka_jelacic/lib/blocs/button_event.dart

```diff
@@ -0,0 +1,11 @@
+import 'package:flutter/cupertino.dart';
+
+abstract class ButtonEvent {}
+
+class ButtonPressEvent extends ButtonEvent {
+  final int index;
+  final String imageUrl;
```

Comment:
Bilo bi bolje da imageUrl ne šalješ na eventu, nego samo u blocu da odrediš na osnovu indexa koja slika treba da se prikaže. Ovako imaš url na eventu, šalješ ga u bloc, vraćaš ga na UI, što je nepotrebno.